toqito/matrices/operator_sinkhorn.py

```diff
@@ -0,0 +1,195 @@
+"""Sinkhorn operator."""
+
+import warnings
+
+import numpy as np
+import scipy as sp
+
+from toqito.channels.partial_trace import partial_trace
+from toqito.matrix_props.is_square import is_square
+
+
+def operator_sinkhorn(
+    rho: np.ndarray, dim: list[int] = None, tol: float = np.sqrt(np.finfo(float).eps), max_iterations: int = 10000
+) -> tuple[np.ndarray, list[np.ndarray]]:
+    r"""Perform the operator Sinkhorn iteration.
+
+    This function implements the iterative Sinkhorn algorithm to find a density matrix
```

Comment:
What is the Sinkhorn algorithm? Describe it briefly. 

Add a reference to this too.